packages/ai-bot/main.ts

```diff
@@ -471,3 +504,54 @@ Common issues are:
   Sentry.captureException(e);
   process.exit(1);
 });
+
+// Handle SIGTERM (sent by ECS when it shuts down the instance)
+process.on('SIGTERM', async () => {
+  await handleShutdown();
+  process.exit(0);
+});
+
+// Handle SIGINT (Ctrl+C from user)
+process.on('SIGINT', async () => {
+  await handleShutdown();
+  process.exit(0);
+});
+
+let waitForActiveGenerationsPromise: Promise<void> | undefined;
+async function handleShutdown() {
+  if (waitForActiveGenerationsPromise) {
+    return waitForActiveGenerationsPromise;
+  }
+
+  isShuttingDown = true;
+
+  log.info('Shutting down...');
+
+  waitForActiveGenerationsPromise = waitForActiveGenerations();
+  await waitForActiveGenerationsPromise;
+  waitForActiveGenerationsPromise = undefined;
+}
+
+async function waitForActiveGenerations() {
+  let minutes = 5;
```

Comment:
This reminds me we need to match the timeout in the infra for process termination timeout